### PR TITLE
Refactor events listening logic and add key registry listener

### PR DESCRIPTION
**Pull request by haardikk21**
- handle ChangeRecoveryAddress event from IdRegistry
- add key registry listener, refactor events listening code

### CODE REVIEW

File: lib/common/build.rs [hunk 1]
```diff
@@ -2,6 +2,8 @@ extern crate prost_build;
 use std::path::PathBuf;
 
 fn main() {
+    println!("cargo:rerun-if-changed=../../protobufs");
```
Comment (avichalp) [inline lib/common/build.rs:5]: do we need the print statement here?

Comment (haardikk21) [reply lib/common/build.rs:5]: Yep, weird rust thing - it was a TIL

https://doc.rust-lang.org/cargo/reference/build-scripts.html

Effectively caches the build unless something in that dir changes

**Comment by avichalp** [comment]
@haardikk21: Looks good to me! I think, we can merge this. I will work on refactoring the event listeners to abstract out the repeated code in there in a different PR.

**Comment by haardikk21** [comment]
> @haardikk21: Looks good to me! I think, we can merge this. I will work on refactoring the event listeners to abstract out the repeated code in there in a different PR.

Sounds good. FWIW, there's some TODO comments to add logic beyond just persisting the event in the key registry listener, and then one more listener left to do iirc. 

I can tackle that later or if you want to, feel free to do that

**Pull request merged**